clients/spark/core/src/main/scala/io/treeverse/gc/UncommittedGarbageCollector.scala

```diff
@@ -68,18 +80,89 @@ object UncommittedGarbageCollector {
                         UncommittedGarbageCollector.UNCOMMITTED_GC_SOURCE_NAME
                        )
     val apiClient = ApiClient.get(apiConf)
-    val uncommittedLocation = args(2) // TODO use actual uncommitted location
-    val outputLocation = args(1)
-    val uncommittedGCRunInfo =
-      new DummyUncommittedAddressLister(uncommittedLocation).listUncommittedAddresses(spark, repo)
-    val uncommittedDF = spark.read.parquet(uncommittedGCRunInfo.uncommitedLocation)
-    val addressesToDelete =
-      getAddressesToDelete(apiClient,
-                           repo,
-                           uncommittedDF,
-                           spark.emptyDataFrame.withColumn("address", lit("")),
-                           DateUtils.addHours(new Date(), -6)
-                          )
-    addressesToDelete.write.parquet(outputLocation)
+    var storageNamespace = apiClient.getStorageNamespace(repo, StorageClientType.HadoopFS)
+    if (!storageNamespace.endsWith("/")) {
+      storageNamespace += "/"
+    }
+
+    try {
+      val dataDF = listObjects(storageNamespace, "data", DateUtils.addHours(new Date(), -6))
+      val uncommittedLocation = args(2) // TODO use actual uncommitted location
+      val uncommittedGCRunInfo =
+        new DummyUncommittedAddressLister(uncommittedLocation).listUncommittedAddresses(spark, repo)
+      runID = uncommittedGCRunInfo.runID
+
+      val uncommittedDF = spark.read.parquet(uncommittedGCRunInfo.uncommittedLocation)
+
+      addressesToDelete = getAddressesToDelete(
+        storageNamespace,
+        dataDF,
+        uncommittedDF,
+        spark.emptyDataFrame.withColumn("address", lit(""))
+      )
+      lastSlice = dataDF.select(col("address")).first().toString()
+    } catch {
+      case _: Throwable => success = false
+    } finally {
+      val markID = "mark-id" // TODO (niro): Add mark ID
+      writeReports(
+        storageNamespace,
+        runID,
+        markID,
+        lastSlice,
+        startTime,
+        success,
+        addressesToDelete
+      )
+
+      spark.close()
+    }
+  }
+
+  def writeReports(
+      storageNamespace: String,
+      runID: String,
+      markID: String,
+      lastSlice: String,
+      startTime: String,
+      success: Boolean,
+      removed: DataFrame
+  ): Unit = {
+    val reportDst = s"${storageNamespace}_lakefs/retention/gc/uncommitted/$runID"
+    writeJsonSummary(reportDst, runID, lastSlice, startTime, success, removed.count())
+
+    removed
+      .withColumn(MARK_ID_KEY, lit(markID))
+      .withColumn(RUN_ID_KEY, lit(runID))
+      .write
+      .partitionBy(MARK_ID_KEY, RUN_ID_KEY)
+      .mode(SaveMode.Overwrite)
+      .parquet(s"$reportDst/deleted.parquet")
```

Comment:
This will delete delete all the previous runs for every new run. You need to use [dynamic partition overwrite](https://kontext.tech/article/1067/spark-dynamic-and-static-partition-overwrite#:~:text=From%20version%202.3.,written%20into%20it%20at%20runtime.).

Reply:
This is written in the run_id prefix - i.e. we created a new df for each run